src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -230,7 +230,18 @@
       <OldBuildVersionFiles Include="$(BuildVersionFilePath)BuildVersions-*.props" Exclude="$(BuildVersionFilePath)%(BuildVersionFileItem.Filename).props" />
     </ItemGroup>
     <Delete Files="@(OldBuildVersionFiles)" TreatErrorsAsWarnings="true"/>
+  </Target>
 
+  <Target Name="GenerateBuiltByString" Condition="'$(OSEnvironment)'!='Windows_NT' AND '$(BuiltByString)'==''">
+    <Exec Command="whoami" StandardOutputImportance="Low" IgnoreExitCode="true" IgnoreStandardErrorWarningFormat="true" ConsoleToMSBuild="true">
```

Comment:
AFAIK they are present in almost all Unix systems already. In case where they are not present, I am still setting the IgnoreExitCode to false so worst case scenario build will still pass and builtbystring would be something like `built by: -`